### ISSUE TITLE
polychromatic-cli need a -g/--get option

### ISSUE THREAD

**Issue by zen2** [polychromatic/polychromatic#370]
**polychromatic-cli** need a **-g/--get option** to get value of an option like **-p/--parameter** one is used to set an option.

It's need-able because actually it's impossible to get a current value even using the -k option because parameters values are represented by colors and it's impossible to catch/grep polychromatic-cli output from bash script or any other language.

Maybe it could be worth value to rename -p/--parameter as -s/--set to stay logical.

**Comment by lah7** [comment]
I agree that there is room for improvement. I'm working on a refactor and the final outcome will make it possible to print the currently selected value/parameter for an option. I was thinking of:

#### Get
```
$ polychromatic-cli -o brightness
Razer BlackWidow Chroma's brightness is currently 50%.
Razer Mamba's brightness is currently 100%.
```

And perhaps if scriptable output is desired:
```
$ polychromatic-cli -o brightness --simple
50% 100%

$ polychromatic-cli -o brightness --simple -d keyboard
50%
```

#### Set
```
$ polychromatic-cli -o brightness -p 75
Razer BlackWidow Chroma's brightness set to 75%.
Razer Mamba's brightness set to 75%.
```

_Subject to change._ I don't think I'll rename `-p / --parameter` to avoid breaking things for existing users, but abstracting the get/set is something I'll need to give a bit more thought, especially considering all the devices.

If you have examples of how you'd like the input/output to look, that would be helpful feedback!

**Comment by zen2** [comment]
I think minimal output is better.
Especially to use it as an api for script.
Need simply to have a real separator without ambiguity: space character can be good if there is no field at all that can contain spaces otherwise we need another separator like a comma.
I have only one device so I can't presume output for several devices.

about --simple option, why not use **-g/--get** and keep -p as **-p/--put** to keep coherency.

**Comment by lah7** [comment]
I presume your script(s) in question are shell/bash (.sh) scripts? If you need require more granular control over the status of your device, you may want to consider writing a Python script (.py) that uses OpenRazer's API directly. See [some examples](https://github.com/openrazer/openrazer/tree/master/examples) and [some outdated docs](https://public.z3ntu.xyz/razer/pylib/). Though I'd suggest tinkering in a `python3` interpreter and using <kbd>Tab</kbd> to see possible functions.

`polychromatic-cli` is designed to be _human-friendly_ for anyone wanting to change their lighting while using the Terminal, or for simple things like a cron schedule. I'm not too sure how to make it work for scripting purposes, especially since there could be many devices involved. After this refactor, I do want to support "getting" the current status, which currently is not possible.

**Comment by zen2** [comment]
In my actual case, python api will be perfect because I develop a tray widget that is already in python but I think a simple way to do it in bash is a must.
Main usage in bash is to change color of keyboard keys or mouse depending on desktop event.
I get used in the past to do it with some simple bash command for Logitech using https://github.com/MatMoul/g810-led.

As the main purpose is to set parameter, to get parameter is more specific to real function like to know the mouse DPI or the mouse polling frequency. If you want to change such parameters, you need to display their state that is easy if you've set it but you can't know if another program have change it or which state you get after a fresh boot.

**Comment by mike-lloyd03** [comment]
Hi just wanted to second support for a `--get` argument in the CLI. I'm writing a script for managing keyboard brightness after being idle and it would be good to be able to read the value before making changes to it.

**Comment by Nqtural** [comment]
Updates?

**Comment by lah7** [comment]
`polychromatic-cli` is deprecated and will be removed in future. You might be interested in [`razer-cli`](https://github.com/lolei/razer-cli) as a potential replacement for command line usage.

There is also OpenRazer's Python library (possibly as a one-liner within Bash) or possibly a program that gets a value directly from the D-Bus `org.razer` session.
